flutter/lib/src/navigation/sentry_navigator_observer.dart

```diff
@@ -86,7 +86,9 @@ class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
   void didPush(Route<dynamic> route, Route<dynamic>? previousRoute) {
     super.didPush(route, previousRoute);
 
-    _setCurrentRoute(route.settings.name);
+    final routeName =
+        (_routeNameExtractor?.call(route.settings) ?? route.settings).name;
+    _setCurrentRoute(routeName);
```

Comment:
We can extract to a method and avoid duplication since `didReplace` does the same.

Reply:
`_startTransaction` should also take `routeName` then.

Reply:
Done.